sidechain/rpc-handler/src/direct_top_pool_api.rs

```diff
@@ -154,14 +155,20 @@ where
 	G: PartialEq + Encode + Decode + Debug + Send + Sync + 'static,
 {
 	debug!("Author submit and watch trusted operation..");
-
 	let hex_encoded_params = params.parse::<Vec<String>>().map_err(|e| format!("{:?}", e))?;
 
 	let request =
 		Request::from_hex(&hex_encoded_params[0].clone()).map_err(|e| format!("{:?}", e))?;
 
 	let shard: ShardIdentifier = request.shard;
 	let encrypted_trusted_call: Vec<u8> = request.cyphertext;
+
+	if encrypted_trusted_call.len() > MAX_TOP_SIZE_TO_ENTER_POOL {
+		let error_msg = "Trusted operation too large";
+		error!("{}", error_msg);
+		return Err(error_msg.into())
+	}
+
```

Comment:
we now expose the rpc not to potential DDoS load which we can't easily proxy away anymore. Therefore we at least limit the size per call before forwarding anything further into the enclave